src/doc/tarpl/drop-flags.md

```diff
@@ -0,0 +1,81 @@
+% Drop Flags
+
+The examples in the previous section introduce an interesting problem for Rust.
+We have seen that's possible to conditionally initialize, deinitialize, and
+*reinitialize* locations of memory totally safely. For Copy types, this isn't
+particularly notable since they're just a random pile of bits. However types
+with destructors are a different story: Rust needs to know whether to call a
+destructor whenever a variable is assigned to, or a variable goes out of scope.
```

Comment:
I don't understand your response.

This `drop-flags.md` section is all about where we might have a conditional invocation of a value's drop code (or if you prefer, "a path's drop code).

I am trying to point out that these flags (which are currently embedded in values, and in the long term will be stack local, but both now and in the long term do need to exist *somewhere*), that these flags exist for nearly any L-value, not just variables.

the way the text is written, it implies (a bit too strongly IMO) that conditional drop is solely about a local variable in its entirety.  But there may be many more drop flags present than the number of local variables.

(The main exception I can think of is that we won't have drop-flags for array-index L-values `a[i]`)